bundles/org.openhab.binding.sunsa/src/main/java/org/openhab/binding/sunsa/internal/device/SunsaDeviceHandler.java

```diff
@@ -0,0 +1,188 @@
+import org.openhab.core.library.types.StopMoveType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SunsaDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author jirom - Initial contribution
+ */
+@NonNullByDefault
+public class SunsaDeviceHandler extends BaseThingHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SunsaDeviceHandler.class);
+
+    private @Nullable SunsaService sunsaService;
+    private @Nullable SunsaDeviceConfiguration config;
+
+    public SunsaDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getConfigAs(SunsaDeviceConfiguration.class);
+        this.sunsaService = ((SunsaCloudBridgeHandler) getBridge().getHandler()).getSunsaService();
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            if (isConnected()) {
+                updateStatus(ThingStatus.ONLINE);
+            } // getState already updates the status with more detail if it thing is not reachable
+        });
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateState();
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_CONFIGURABLE_POSITION: {
+                if (command instanceof StopMoveType) {
+                    // We don't support this
+                    LOGGER.warn("Received {} command which is not supported and is explicitly ignored.", command);
+                    return;
+                }
+
+                final int rawPosition = getPositionAdapter(channelUID.getId())
+                        .getRawPosition(getPositionInterpolationFactor(command));
+                LOGGER.debug("Attempting update to raw position to: {}", rawPosition);
+                updatePosition(rawPosition);
+                break;
+            }
+
+            case CHANNEL_RAW_POSITION: {
+                final int rawPosition = ((DecimalType) command).intValue();
+                LOGGER.debug("Attempting update to raw position to: {}", rawPosition);
```

Comment:
Not sure if the logging should be here. As the framework allready logs the event. Same for other switch options